test/server/cards/06_SEC/units/HiredSlicer.spec.ts

```diff
@@ -0,0 +1,165 @@
+                        deck: ['tieln-fighter', 'imperial-interceptor', 'snowspeeder', 'cell-block-guard', 'daring-raid']
+                    }
+                });
+            });
+
+            it('should allow revealing top 2 cards of own deck, exhausting a unit sharing a trait, and putting cards on bottom in random order', function () {
+                const { context } = contextRef;
+
+                // Attack with Hired Slicer
+                context.player1.clickCard(context.hiredSlicer);
+                context.player1.clickCard(context.p2Base);
+
+                // Attack declared, now On Attack ability triggers
+                expect(context.player1).toHavePrompt('Select one');
+                expect(context.player1).toHaveExactPromptButtons(['Your deck', 'Opponent\'s deck', 'Pass']);
+                context.player1.clickPrompt('Your deck');
+
+                // Top 2 cards are Battlefield Marine (Trooper) and Pyke Sentinel (Underworld)
+                expect(context.getChatLogs(2)).toContain('player1 uses Hired Slicer to reveal Battlefield Marine and Pyke Sentinel');
+
+                // Can exhaust units that share traits with revealed cards
+                // Wampa has Creature trait, Death Star Stormtrooper has Trooper, Enfys Nest has Underworld
+                expect(context.player1).toBeAbleToSelectExactly([context.deathStarStormtrooper, context.enfysNestMarauder]);
+                context.player1.clickCard(context.deathStarStormtrooper);
+
+                expect(context.deathStarStormtrooper.exhausted).toBe(true);
+
+                // Check that cards were put on bottom of deck
+                expect(context.player1.deck.length).toBe(5);
+                const bottomTwoCards = context.player1.deck.slice(-2);
+                expect(bottomTwoCards).toContain(context.battlefieldMarine);
+                expect(bottomTwoCards).toContain(context.pykeSentinel);
+            });
+
+            it('should allow revealing top 2 cards of opponent\'s deck', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.hiredSlicer);
+                context.player1.clickCard(context.p2Base);
+
+                expect(context.player1).toHavePrompt('Select one');
+                context.player1.clickPrompt('Opponent\'s deck');
+
+                // Top 2 cards are TIE/LN Fighter (Imperial, Vehicle) and Imperial Interceptor (Imperial, Vehicle)
+                // Note: Chat log uses lowercase 'ln', not 'LN'
+                expect(context.getChatLogs(2)).toContain('player1 uses Hired Slicer to reveal TIE/ln Fighter and Imperial Interceptor');
+
+                // Death Star Stormtrooper has Imperial trait, so it CAN be exhausted!
+                expect(context.player1).toBeAbleToSelectExactly([context.deathStarStormtrooper]);
+                context.player1.clickCard(context.deathStarStormtrooper);
```

Comment:
Just to verify the target filter applies to all units in play could we add the following to the test setup:
- A _friendly_ unit that shares a trait (just for one of these cases is okay, doesn't have to be both) with a revealed card
- A deployed leader that shares a trait